provider/pkg/gen/schema_test.go

```diff
@@ -462,3 +464,134 @@ func TestIsApiVersionQueryParam(t *testing.T) {
+		assert.Equal(t, `This is a wonderful resource.
+
+Azure REST API version: 2022-02-22. In version 1.x of the Azure Native provider, it used API version 2021-01-01.
+
+Other available API versions: 2023-03-03, 2024-04-04.
+
+Additional docs.`, strings.TrimSpace(desc))
+	})
+}
+
+func TestFormatDescriptionV3(t *testing.T) {
+	t.Parallel()
+
+	g := &packageGenerator{
+		majorVersion: 3,
+		versioning:   versioningStub{},
+	}
+
+	t.Run("no description", func(t *testing.T) {
+		desc := g.formatDescription("", "SomeResource", "2022-02-22", "", nil)
+		assert.Equal(t, "Azure REST API version: 2022-02-22.", strings.TrimSpace(desc))
+	})
+
+	t.Run("description", func(t *testing.T) {
+		desc := g.formatDescription("This is a wonderful resource.", "SomeResource", "2022-02-22", "", nil)
+		assert.Equal(t, "This is a wonderful resource.\n\nAzure REST API version: 2022-02-22.", strings.TrimSpace(desc))
+	})
+
+	t.Run("additional docs", func(t *testing.T) {
+		desc := g.formatDescription("", "SomeResource", "2022-02-22", "", pulumi.StringRef("Additional docs."))
+		assert.Equal(t, "Azure REST API version: 2022-02-22.\n\nAdditional docs.", strings.TrimSpace(desc))
+	})
+
+	t.Run("previous default version", func(t *testing.T) {
+		desc := g.formatDescription("", "SomeResource", "2022-02-22", "2021-01-01", nil)
+		assert.Equal(t, "Azure REST API version: 2022-02-22. In version 2.x of the Azure Native provider, it used API version 2021-01-01.", strings.TrimSpace(desc))
+	})
+
+	t.Run("other versions", func(t *testing.T) {
+		g := &packageGenerator{
+			majorVersion: 3,
+			moduleName:   "storage",
+			versioning: versioningStub{
+				getAllVersions: func(moduleName openapi.ModuleName, resource string) []openapi.ApiVersion {
+					return []openapi.ApiVersion{"2023-03-03", "2024-04-04"}
+				},
+			},
+		}
+
+		desc := g.formatDescription("", "SomeResource", "2022-02-22", "", nil)
+		assert.Equal(t, `Azure REST API version: 2022-02-22.
+
+Other available API versions: 2023-03-03, 2024-04-04. These can be accessed by generating a local SDK package using the CLI command `+"`"+`pulumi package add azure-native storage [ApiVersion]`+"`"+`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.`,
+			strings.TrimSpace(desc))
+	})
+
+	t.Run("everything", func(t *testing.T) {
+		g := &packageGenerator{
+			majorVersion: 3,
+			moduleName:   "Authorization",
+			versioning: versioningStub{
+				getAllVersions: func(moduleName openapi.ModuleName, resource string) []openapi.ApiVersion {
+					return []openapi.ApiVersion{"2023-03-03", "2024-04-04"}
+				},
+			},
+		}
+
+		desc := g.formatDescription("This is a wonderful resource.", "SomeResource", "2022-02-22", "2021-01-01", pulumi.StringRef("Additional docs."))
+		assert.Equal(t, `This is a wonderful resource.
+
+Azure REST API version: 2022-02-22. In version 2.x of the Azure Native provider, it used API version 2021-01-01.
```

Comment:
```suggestion
Uses Azure REST API version: 2022-02-22. In version 2.x of the Azure Native provider, it used API version 2021-01-01.
```
Nit. I think making the first a complete sentence makes the second sentence read better.